pkg/exchange/bybit/bybitapi/get_k_lines_request.go

```diff
@@ -0,0 +1,107 @@
+package bybitapi
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/c9s/requestgen"
+
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+//go:generate -command GetRequest requestgen -method GET -responseType .APIResponse -responseDataField Result
+//go:generate -command PostRequest requestgen -method POST -responseType .APIResponse -responseDataField Result
+
+type IntervalSign string
+
+const (
+	IntervalSignDay   IntervalSign = "D"
+	IntervalSignWeek  IntervalSign = "W"
+	IntervalSignMonth IntervalSign = "M"
+)
+
+type KLinesResponse struct {
+	Symbol string `json:"symbol"`
+	// An string array of individual candle
+	// Sort in reverse by startTime
+	List     []KLine  `json:"list"`
+	Category Category `json:"category"`
+}
+
+type KLine struct {
+	// list[0]: startTime, Start time of the candle (ms)
+	StartTime types.MillisecondTimestamp
+	// list[1]: openPrice
+	Open fixedpoint.Value
+	// list[2]: highPrice
+	High fixedpoint.Value
+	// list[3]: lowPrice
+	Low fixedpoint.Value
+	// list[4]: closePrice
+	Close fixedpoint.Value
+	// list[5]: volume, Trade volume. Unit of contract: pieces of contract. Unit of spot: quantity of coins
+	Volume fixedpoint.Value
+	// list[6]: turnover, Turnover. Unit of figure: quantity of quota coin
+	TurnOver fixedpoint.Value
+}
+
+const KLinesArrayLen = 7
+
+func (k *KLine) UnmarshalJSON(data []byte) error {
+	var jsonArr []json.RawMessage
+	err := json.Unmarshal(data, &jsonArr)
+	if err != nil {
+		return err
+	}
+	if len(jsonArr) != KLinesArrayLen {
+		return fmt.Errorf("unexpected K Lines array length: %d, exp: %d", len(jsonArr), KLinesArrayLen)
+	}
+
+	err = json.Unmarshal(jsonArr[0], &k.StartTime)
+	if err != nil {
+		return err
+	}
+
+	values := make([]fixedpoint.Value, len(jsonArr)-1)
+	for i, j := range jsonArr[1:] {
+		err = json.Unmarshal(j, &values[i])
+		if err != nil {
+			return err
```

Comment:
When an error is returned, the first question will be which field is broken. So it would be helpful if the error includes the invalid index.